kbs/src/attestation/intel_trust_authority/mod.rs

```diff
@@ -7,14 +7,41 @@ use crate::token::{
     jwk::JwkAttestationTokenVerifier, AttestationTokenVerifier, AttestationTokenVerifierConfig,
     AttestationTokenVerifierType,
 };
+use crate::attestation::{generic_generate_challenge, make_nonce};
 use anyhow::*;
 use async_trait::async_trait;
+use attestation_service::HashAlgorithm;
 use base64::{engine::general_purpose::STANDARD, Engine};
+use kbs_types::Challenge;
 use kbs_types::{Attestation, Tee};
+use lazy_static;
 use reqwest::header::{ACCEPT, CONTENT_TYPE};
 use serde::{Deserialize, Serialize};
 use serde_json::json;
 
+lazy_static! {
+    /// The hash algorithm used for Intel SGX.
+    static ref SGX_HASH_ALGORITHM: String = HashAlgorithm::Sha256.to_string().to_lowercase();
+
+    /// The hash algorithm used for Intel TDX.
+    static ref TDX_HASH_ALGORITHM: String = HashAlgorithm::Sha512.to_string().to_lowercase();
+
+    static ref ERR_NO_TDX_ALGORITHM: String = format!(
+        "Intel Trust Authority: {:?} not supported by TDX TEE",
+        *TDX_HASH_ALGORITHM
+    );
+    static ref ERR_NO_SGX_ALGORITHM: String = format!(
+        "Intel Trust Authority: {:?} not supported by SGX TEE",
+        *TDX_HASH_ALGORITHM
+    );
```

Comment:
We might not need these consts. For `SGX_HASH_ALGORITHM`, we can 
```rust
enum HashAlgorithm {
    #[strum(serialize = "sha256")]
    Sha256,
    ...
}

...
if supported_hash_algorithms.contains(HashAlgorithm::Sha256.as_ref()) {
...
}
```

For errors, I suggest that directly use `.context("...")` for extra error context.